subjects/corewar/README.md

```diff
@@ -0,0 +1,366 @@
+- The VM assumes the binary is in big-endian and read it accordingly.
+
+Those are the cases were a file is considered corrupted:
+
+- Wrong signature code.
+- Declared size of the program not matching the actual size.
+- The size of the program is bigger that the maximum allowed size.
+- The total file size is smaller than the minimum size.
+
+### Greeting and end game messages
+
+At the start of each game the VM should print:
+
+```
+For this match the players will be:
+Player 1 ([X] bytes): [NAME] ([DESCRIPTION])
+Player 2 ([X] bytes): [NAME] ([DESCRIPTION])
+...
+```
+
+At the end of the game the vm should print `cycle [X]: The winner is player [X]: [NAME]!`.
+
+> If nobody executed a valid `live` statement the end message should be `cycle [X]: Nobody wins!`.
+
+#### Circular memory space of the arena
+
+The memory where the players will fight is circular, this means if we want to move forward from the last address in memory (for example `4095`) we will arrive at address `0`.
+
+It implies that moving backward of one position from `0` will bring us to the address `4095`.
+
+> Having circular memory guarantees players they will never overflow the memory space they are playing in.
+
+#### Stop player execution
+
+Every `CYCLE_TO_DIE` the VM will check for every player if it signaled it was alive at least once, if it didn't all processes created by this player will be immediately killed.
+
+To avoid infinite games, `CYCLES_TO_DIE` will be decremented by `CYCLE_DELTA` under certain conditions:
+
+- If during the last life loop there were at least `NBR_LIVE` successfully executed by the players.
+- If it has been `MAX_CHECKS` life loops since it was decremented last time.
+
+> Notice that some smart players may trick another player to start making `live` statements for them, this virtually means a player can still make `live` statements after all its processes were killed.
+
+### The Assembly language
+
+In the assembly file there must be a `.name "[Name of the player]"` field and a `.description "[Description of the player]"` before any instruction.
+
+Every instruction has to be on a single line and may be preceded by a label declaration.
+The instruction name is separated from the optional label and the parameters by one or more spaces.
+
+The parameters are separated from each other by `,`.
+
+The instructions will be of different sizes depending on the number of type of parameters passed.
+
+#### Labels
+
+Labels are also called pseudo instructions, in the sense they won't be translated into binary code.
+They are widely used to help programmers creating assembly code without needing to remember and count relative memory positions.
+
+So when looking in the binaries the labels will be replaced in order to match their relative position in bytes from the place they were used.
+
+As an example in the `ameba.s` player `zjmp %:hello` is exactly the same of `zjmp %-5` (`-5` being `ff fb` in hexadecimal notation).
```

Comment:
`ameba.s` is not introduced before this point - it might be easier to have the `ameba.s` example right after the section **### The Assembly language*** or at least a reference to jump to the example snippet

Reply:
it is true this part is not linear, still `ameba` will be present in the playground and it is also fully translated and explained later in the subject.

To delay this part means having less info when the labels are explained or to explain labels at a very later point (where they would have shown up already multiple times, reproducing the same issue at a bigger scale).

Still there may be other ways to go, if you feel we can rephrase or move just part of those paragraphs let me know, I don't have strong opinions on that one